### PR TITLE
Feature/task 4

### PR DESCRIPTION
**Pull request by LindaH32**

### CODE REVIEW

**Comment by Suzii** [comment]
I love the commit messages :+1: Review comming soon ;)

File: onboarding-task/src/components/ItemRead.tsx [hunk 1]
```diff
@@ -1,31 +1,33 @@
+///<reference path="../../node_modules/@types/react/index.d.ts"/>
```
Comment (PetrSvirak) [inline onboarding-task/src/components/ItemRead.tsx]: This is not necessary, remove the line (global definitions should be applied everywhere by, well, definition)

File: onboarding-task/src/components/ItemRead.tsx [hunk 2]
```diff
@@ -1,31 +1,33 @@
...
+import { IItemViewModel } from '../models/IItemViewModel';
+import { IAction } from '../actionCreators/IAction';
 
 interface IItemReadDataProps {
   item: IItemViewModel;
 }
 
 interface IItemReadCallbackProps {
-  onDoubleClick: () => IAction;
+  onClick: () => IAction;
```
Comment (PetrSvirak) [inline onboarding-task/src/components/ItemRead.tsx]: This was completely unnecessary, please re-read [my comment on the topic](https://github.com/Suzii/kentico-onboarding-js/pull/45#discussion_r112549692) and concentrate on the `focus` part (you need not to revert the change from double click to (single) click – it is completely up to you, though, what your preferred behavior of the item is).

File: onboarding-task/test/modelTests/item.test.js [hunk 3]
```diff
@@ -1,25 +1,43 @@
...
       isEdited: false,
     });
+
     const tested = item.toObject();
 
     expect(tested).toEqual(expected);
   });
-  it('Creates correct itemViewModel', () => {
+
+  it('Works the with() method correctly', () => {
```
Comment (PetrSvirak) [inline onboarding-task/test/modelTests/item.test.js]: Not much English here :wink:

File: onboarding-task/package.json [hunk 4]
```diff
@@ -61,6 +61,7 @@
     "whatwg-fetch": "1.0.0"
   },
   "dependencies": {
+    "@types/prop-types": "^15.5.1",
```
Comment (PetrSvirak) [inline onboarding-task/package.json]: These are only typings for the `prop-types` (and you already have them in the `dev-dependencies`), not `prop-types` themselves. Type definitions (`d.ts`) files do never make sense in `dependencies` as TS is stripped from your code to make it plain JS before passed to the browser.

File: onboarding-task/src/components/ItemEdit.tsx [hunk 5]
```diff
@@ -51,7 +57,7 @@ class ItemEdit extends React.PureComponent<TItemProps, IItemEditState> {
     return (
       <div>
         <span className="form-inline">{this.props.item.index}.
-          <input className="form-control" value={this.state.text} onChange={this._handleOnChange} />
+          <input ref={(input) => {this.input = input; }} className="form-control" value={this.state.text} onChange={this._handleOnChange} />
```
Comment (PetrSvirak) [inline onboarding-task/src/components/ItemEdit.tsx]: This all construction is really nice. I like the way you used `ref` and that you did not use a `string` constant for it. It is also typed soundly. However, it is actually enough to put `autoFocus` attribute to the `input` (see this [SO](http://stackoverflow.com/a/28891210/1138663)).

File: onboarding-task/test/App.test.js [hunk 6]
```diff
@@ -2,7 +2,7 @@ import React from 'react';
 import ReactDOM from 'react-dom';
 import App from '../src/App.jsx';
 
-it('renders without crashing', () => {
+xit('renders without crashing', () => {
```
Comment (PetrSvirak) [inline onboarding-task/test/App.test.js]: It is OK to ignore this very test, however, there should a comment on why is the test ignore. Please, write one.

**Comment by PetrSvirak** [comment]
All's good now, thanks :)
